src/fparser/two/tests/test_fortran2003.py

```diff
@@ -3528,30 +3528,11 @@ def test_format_item_list():  # R1002, R1003
 #
 
 
-def test_main_program():  # R1101
-
-    tcls = Main_Program
-    obj = tcls(get_reader('''\
-program a
-end
-    '''))
-    assert isinstance(obj, tcls), repr(obj)
-    assert str(obj) == 'PROGRAM a\nEND PROGRAM a'
-
-    obj = tcls(get_reader('''\
-program a
-  real b
-  b = 1
-  contains
-  subroutine foo
-  end
-end
-    '''))
-    assert isinstance(obj, tcls), repr(obj)
-    assert (str(obj) ==
-            'PROGRAM a\n  REAL :: b\n  b = 1\n  CONTAINS\n  SUBROUTINE foo\n'
-            '  END SUBROUTINE foo\nEND PROGRAM a')
+def test_main_program0():  # R1101 helper
```

Comment:
Is this not a test in its own right? `test_main_program0` tests the class `Main_Program0`.

However, in the longer term I agree that the Main_Program0 class should be merged with Main_Program if at all possible which would allow us to merge the tests as well. My understanding is that it is only treated separately because the base class can't cope with an optional first line and a mandatory last line.